tron/core/actionrun.py

```diff
@@ -1306,7 +1344,8 @@ def recover(self) -> Optional[KubernetesTask]:
             log.error(f"{self} no task ID, cannot recover")
             self.fail_unknown()
             return None
-
+        # when we get attempts here, the field_selector_env = {'PAASTA_POD_IP': ['status.podIP']} which is in a diff format than
+        # the field_selector_env in submit_command function.
```

Comment:
I've noticed that the format for the ``field_selector_env`` in the recover function (which we get from the self object) is in the form of 
```
{'PAASTA_POD_IP': ['status.podIP']}
```
and this is different than the format in the submit_command function (which we get here from the attempt argument) from what i recall it was in the right format here as a dict of str, ConfigFieldSelectorSource. Thus, when reading I always make sure i convert to this object as in 
```
                "field_selector_env": {
                    key: ConfigFieldSelectorSource(**val) for key, val in json_data["field_selector_env"].items()
                },
```

Reply:
might be worth moving this comment closer to where we do this - it feels a tad out of place here since we're not doing anything special with `field_selector_env` in this function :)